spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerOperations.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.secretmanager;
+
+import com.google.protobuf.ByteString;
+
+/**
+ * Describes supported operations that one can perform on the Secret Manager API.
+ *
+ * @author Daniel Zou
+ * @since 1.3
+ */
+public interface SecretManagerOperations {
+
+	/**
+	 * Creates a new secret using the provided {@code secretId} and creates a new version of
+	 * the secret with the provided {@code payload}.
+	 *
+	 * <p>
+	 * If there is already a secret saved in SecretManager with the specified
+	 * {@code secretId}, then it simply creates a new version under the secret with the secret
+	 * {@code payload}.
+	 *
+	 * @param secretId the secret ID of the secret to create.
+	 * @param payload the secret payload string.
+	 */
+	void createSecret(String secretId, String payload);
```

Comment:
I wonder if `setSecret` is more precise, since the secret could be created _or_ updated.

Reply:
Ah, so in Secret Manager previous versions of secrets are never deleted; i.e. the `Secret` is like a container/parent, and then you create `SecretVersions` under the Secret. I think "create" would be correct, but not sure if there may be better choices for what is being created; like do you feel `createSecretVersion` may be more clear?

Reply:
I think it's okay to leave it as is.
I thought about `createSecretVersion` but that takes focus off "secret" and onto "version", which is also not ideal.

Reply:
Sounds good, will leave as is.